### DISCUSSION TITLE
sd_mp3 going back one directory/how to set perticular file in source object

### DISCUSSION THREAD

**Discussion by RitikDutta** [pschatzmann/arduino-audio-tools#248] [Q&A] [answered]
hey there, 
I am trying to modify play mp3/wav on sdcard in audiokit example
every thing working fine
ther's one question

i want to play specific sound in two different conditions.

like play sample1.mp3 when condition 1 true
play sample2.mp3 when condition2 is true

i am unable to find a way to point files in source object ,, maybe this needs to be directory only
AudioSourceSdFat source("/", "wav", PIN_AUDIO_KIT_SD_CARD_CS, speedMz);
so i done 1 trick i created two directory (sample1, sample2) with their .mp3 files respectivly
then i set 
AudioSourceSdFat source("/sample1", "wav", PIN_AUDIO_KIT_SD_CARD_CS, speedMz);
in first condition and player.begin to play the audio
and similarly
AudioSourceSdFat source("/sample2", "wav", PIN_AUDIO_KIT_SD_CARD_CS, speedMz);
in second..

it is working fine but only one time for either of the condition because,
first it goes in any condition say condition1 is true so it goes to sample1 directory and plays the sound in current directory
but when second condition match it is searching sample2 directory in sample1(which is current directory now) and gives no file found error

so i want solution for 
- how to go one directory back ( i have tried ".." , "./" but not working)
- or how to set file path(not destination path) in source object:
     AudioSourceSdFat source("/", "wav", PIN_AUDIO_KIT_SD_CARD_CS, speedMz);

**Comment by pschatzmann** [comment, marked as answer]
Why don't you just use the regular stream API: on condition 1 you copy the file1 to the EncodedAudioStream() and on condition 2 you copy the other file.
Another way - if you want to use the AudioPlayer class - would be that you implement your own AudioSource subclass.

**Reply by RitikDutta** [reply]
thankyou for the quick solution, i am gone through the AudioSource class, find the solution there
source.setFileFilter("*sample1*"); done the job

**Comment by pschatzmann** [comment]
Cool, I completely forgot that I have added that method...
